src/common/io.py

```diff
@@ -1593,23 +1610,29 @@ def get_variables_data(self,
         if isinstance(start_index, int) and isinstance(stop_index, int) and stop_index < start_index:
             raise ValueError(f"Invalid values for {start_index=} and {stop_index=}, " + \
                               "'start_index' needs to be less than or equal to 'stop_index'.")
-        trajectories = []
 
-        # Get the time trajectory
+        trajectories = {}
+
+        # Get the corresponding time trajectory
         if not self._contains_diagnostic_data:
             time = self._get_trajectory(0, start_index, stop_index)
         else:
             # Since we interpolate data if diagnostics is enabled
             time = self._get_diagnostics_trajectory(0, start_index, stop_index)
 
-        # Need to account for data that might be added while we are iterating over 'names' later
-        if stop_index is None:
-            stop_index = len(time) + start_index
+        # If stop_index > number of data points, and data gets added while we are iterating
+        # then we might get trajectories of unequal lengths. Therefore ensure we set stop_index here accordingly.
+        stop_index = min(len(time) + start_index, float('inf') if stop_index is None else stop_index)
```

Comment:
@chria Added the fix we discussed. Didn't add a test, I don't see how we can test this since I need to add data when this function is being called, in fact after a particular line has been called, in this function. Let me know if you have any ideas otherwise.

Reply:
Great that you added it! Good with the comment. However, it is a bit confusing that you here use a "float" instead of an int. I'd recommend to change it so that we do use ints. A simple approach would be (similar to what you had before):
max_index = len(time) + start_index)
if stop_index is None:
    stop_index = max_index
else:
   stop_index = min(max_index, stop_index)

Reply:
Fixed in 81fe25e

Reply:
Looking at this again - I get a bif confused. Lets say that we have 10 points in the result file. We set stop_index to None and start_index to 10. What value would stop_index get after this if-statement? Would it not get 20? (i.e. 10+10) which is more than what is in the file?

Reply:
No this will not be a problem since before this if-statement we have:
```
# Get the corresponding time trajectory
if not self._contains_diagnostic_data:
    time = self._get_trajectory(0, start_index, stop_index)
else:
    # Since we interpolate data if diagnostics is enabled
    time = self._get_diagnostics_trajectory(0, start_index, stop_index)

```
Here `len(time)` will be `0` since `start_index=10`. Note we also have tests for this here:
https://github.com/modelon-community/PyFMI/pull/272/files#diff-26824e205001fe146a1634cf74c1dda1cdb8a07b6216545990b773a12da705d4R2028

Reply:
I see, thank you.